src/log/syslog.rs

```diff
@@ -19,7 +19,32 @@ impl Log for Syslog {
         };
 
         let message = format!("{}", record.args());
-        syslog(priority, libc::LOG_AUTH, &message);
+        let message_len = message.bytes().len();
+
+        let mut x: usize = 960;
+        let mut x_prev: usize = 0;
+
+        loop {
+            // floor_char_boundary is currently unstable
+            while !message.is_char_boundary(x) {
```

Comment:
I'm a bit worried about performance here as this `while` loop will repeat itself a lot of times for messages that happen to be shorter than 960. As an example see [this snippet](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d31d7e9260fb3c7488351ee7f63844de) running.

I think this could be easily fixed by adding a conditional that calls `syslog(priority, libc::LOG_AUTH, &message);` and returns earlier if `message_len < 960`